python_modules/dagster/dagster/_core/definitions/auto_materialize_rule.py

```diff
@@ -629,6 +633,107 @@ def unpack(
+class SkipOnNotAllParentsUpdatedRule(
+    AutoMaterializeRule,
+    NamedTuple(
+        "_SkipOnNotAllParentsUpdatedRule", [("require_update_on_all_parent_partitions", bool)]
+    ),
+):
+    """An auto-materialize rule that enforces that an asset can only be requested for execution
+    if all parents have been materialized since its last materialization.
+
+    Properties:
+        require_update_on_all_parent_partitions (Optional[bool]): If true, requires all upstream
+            partitions to be materialized in order for a given asset to update. If false, requires
+            at least one upstream partition in each parent asset to be materialized in order for
+            a given asset to update. Defaults to false.
+    """
+
+    def __new__(cls, require_update_on_all_parent_partitions: Optional[bool] = False):
+        return super(SkipOnNotAllParentsUpdatedRule, cls).__new__(
+            cls,
+            check.bool_param(
+                require_update_on_all_parent_partitions, "require_update_on_all_parent_partitions"
+            ),
+        )
+
+    @property
+    def decision_type(self) -> AutoMaterializeDecisionType:
+        return AutoMaterializeDecisionType.SKIP
+
+    def evaluate_for_asset(
+        self,
+        context: RuleEvaluationContext,
+    ) -> RuleEvaluationResults:
+        asset_partitions_by_waiting_on_asset_keys = defaultdict(set)
+        for candidate in context.candidates:
+            # find the root cause of why this asset partition's parents are outdated (if any)
+
+            parent_partitions = context.asset_graph.get_parents_partitions(
+                context.instance_queryer,
+                context.instance_queryer.evaluation_time,
+                context.asset_key,
+                candidate.partition_key,
+            ).parent_partitions
+
+            updated_parent_partitions = (
+                context.instance_queryer.get_updated_parent_asset_partitions(
+                    candidate,
+                    parent_partitions,
+                    context.daemon_context.respect_materialization_data_versions,
+                )
+                | set().union(
+                    *[
+                        context.will_materialize_mapping.get(parent, set())
+                        for parent in context.asset_graph.get_parents(context.asset_key)
+                    ]
+                )
+            )
+
+            if self.require_update_on_all_parent_partitions:
+                # All upstream partitions must be updated in order for the candidate to be updated
+                non_updated_parent_keys = {
+                    parent.asset_key for parent in parent_partitions - updated_parent_partitions
+                }
+            else:
+                # At least one upstream partition in each upstream asset must be updated in order
+                # for the candidate to be updated
+                upstream_parent_partitions_by_asset_key = _get_asset_partitions_by_asset_key(
```

Comment:
seems like we can avoid this call, as we really just need to see if the set of asset_keys in updated_parent_partitions_by_asset_key equals the set of parent asset keys (`context.asset_graph.get_parents(context.asset_key)`). This saves us the (potentially mildly expensive) work of reorganizing all of the parent partitions into a dictionary.